### PR TITLE
Remove Document from runtime path cache after it was deleted

### PR DESCRIPTION
**Pull request by jdreesen**
In [`Document::getByPath()`](https://github.com/pimcore/pimcore/blob/10.x/models/Document.php#L217-L239) the Document gets cached (unlike the same method on `Assets` and `DataObjects`) but the cache is never reset. So when the was deleted, `getByPath()` still returns the now non-existing object.

```php
$doc = Document::getByPath('/some/path');
$doc->delete();

// still returns the doc!
$doc = Document::getByPath('/some/path');
```

## Changes in this pull request

This PR resets the path cache when the document gets deleted.

### CODE REVIEW

File: models/Document.php [hunk 1]
```diff
@@ -827,6 +827,7 @@ public function delete()
 
         //clear document from registry
         \Pimcore\Cache\Runtime::set(self::getCacheKey($this->getId()), null);
+        \Pimcore\Cache\Runtime::set('document_path_' . md5($this->getRealFullPath()), null);
```
Comment (jdreesen) [inline models/Document.php]: Should we create a `getPathCacheKey($path)` method?

Comment (dvesh3) [reply models/Document.php]: yes, please refactor it to `getPathCacheKey`

Comment (jdreesen) [reply models/Document.php]: done

File: models/Document.php [hunk 2]
```diff
@@ -24,6 +24,7 @@
 use Pimcore\Model\Document\Hardlink\Wrapper\WrapperInterface;
 use Pimcore\Model\Document\Listing;
 use Pimcore\Model\Element\ElementInterface;
+use Pimcore\Model\Element\Service;
```
Comment (dvesh3) [inline models/Document.php]: unused?

Comment (jdreesen) [reply models/Document.php]: oops!

**Comment by jdreesen** [comment]
Do we have to consider the `update()` and `save()` methods here, too? They also update the "id cache".

**Comment by dvesh3** [comment]
> Do we have to consider the `update()` and `save()` methods here, too? They also update the "id cache".

Good catch! I would do it otherwise would not see latest changes on subsequent calls to `getByPath()`

**Review by dvesh3 — APPROVED** [review]

**Comment by dvesh3** [comment]
@jdreesen thanks!

**Comment by brusch** [comment]
@jdreesen this is probably also relevant for assets and data objects, right?

**Comment by jdreesen** [comment]
No, because they don't cache `getByPath()` calls.

**Pull request merged**